### PR TITLE
 Add Integration spec

### PR DESCRIPTION
**Pull request by DaveZag**
# Add Integration spec
### On this branch we:
- [x] **Fixed the **n+1** problem**

Before adding `includes`
before adding

After adding `includes`
after adding it

- [x] **Add test for all the views**

### CODE REVIEW

File: spec/system/post_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
+
```
Comment (1GORDON) [inline spec/system/post_index_spec.rb:2]: Great job @DaveZag 🤝

- [x] As stated in the project requirements, could you implement all the tests as displayed in the screenshot below?
post-index

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @DaveZag ✋

Good job so far!

## Highlights!
- Followed Gitflow ✔️ 
- Good commit messages ✔️
- Linters are passing ✔️
- Descriptive Pull request ✔️
- Professional readme ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions

- [ ] I would suggest you always maintain the same pull request that previous reviewers left changes so that it can help the next reviewers to conduct better and more efficient reviews.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/system/post_index_spec.rb (hunk 1 above)
Comment (DaveZag) [reply spec/system/post_index_spec.rb:2]: @1GORDON Thanks for the feedback. The required changes have been implemented.🙂

**Review by 1GORDON — APPROVED** [review]
# Status: APPROVED 🟢 

Hi @DaveZag ✋

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! :tada: :clap: :clap: :100:

## Highlights!
- Followed Gitflow ✔️
- Good commit messages ✔️
- Linters are passing ✔️
- Descriptive Pull request ✔️
- Professional readme ✔️

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**